src/app/node-version.spec.mjs

```diff
@@ -0,0 +1,29 @@
+/**
+ * When we build the site for deployment to AWS S3 and AWS ECR/ECS we need to use a consistent node version,
+ * partly for compatibility with our code, but also to make sure that all generated js chunk checksums match up
+ * so that the two builds will work together as one app and not fall over with 404 errors as we saw in DON-1073
+ */
+
+import { test } from "node:test";
+import assert from "node:assert/strict";
+import { fileURLToPath } from "node:url";
+import { dirname, resolve } from 'path'
+import {readFileSync} from "node:fs";
+const expectedNodeVersion = '20.13.1';
+
+const modulePath = dirname(fileURLToPath(import.meta.url));
+
+test('should only have the expected node version mentioned in circle config for S3 build', async () => {
+  const circleConfig = readFileSync(resolve(modulePath, '../../.circleci/config.yml')).toString();
+  const nodeMentions = [...circleConfig.matchAll(/node[0-9\\.]+/g)];
```

Comment:
```suggestion
  const nodeMentions = [...circleConfig.matchAll(/node:[0-9\\.]+/g)];
```

Might be confused but I think that only the old cache key had `node` and numbers without a colon separator?

Reply:
@NoelLH you're right. Fixed in d062c5d2 . I also added an assert that there's at least one match in each file.